sdk/eventhub/event-hubs/src/checkpointManager.ts

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { PartitionContext } from "./partitionContext";
+import { EventData } from "./eventData";
+import { PartitionManager } from "./eventProcessor";
+
+/**
+ * Used by createCheckpoint in PartitionManager
+ **/
+export interface Checkpoint {
+  eventHubName: string;
+  consumerGroupName: string;
+  instanceId: string;
+  partitionId: string;
+  sequenceNumber: number;
+  offset: number;
+}
+
+/**
+ * CheckPointManager is created by the library & passed to user's code to let them create a checkpoint
+ */
+export class CheckpointManager {
+  private _partitionContext: PartitionContext; // for internal use by createCheckpoint
+  private _partitionManager: PartitionManager; // for internal use by createCheckpoint
+
+  get partitionContext(): PartitionContext {
```

Comment:
`partitionContext` and `partitionManager` shouldn't be public for now. Customers will have access to both outside of `CheckpointManager`, and we want to the public API as simple as possible to start with.

Reply:
I added these  public getter for now because build was failing with this error: `_partitionContext' is declared but its value is never read.`

I'll remove these getters later after the actual methods implemention.

Reply:
Ah gotcha. You could also just comment them out until we need them too :-)